File: Algorithm/Programmers/Level1/solution(12).py
# 같은 숫자는 싫어
# 정확성, 효율성 모두 통과!

def solution(arr):
    answer = []
    answer.append(arr[0])
    for i in range(1,len(arr)):
        if answer[-1] != arr[i]:
            answer.append(arr[i])
    return answer

File: Algorithm/Programmers/Level1/solution(12).java
// 같은 숫자는 싫어

import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        int[] answer = {};
        ArrayList<Integer> arrlist = new ArrayList<Integer>();
        int temp = -1;
        for (int i = 0; i < arr.length; i++){
            if (temp != arr[i]){
                arrlist.add(arr[i]);
                temp = arr[i];
            }
        }
        
        int n = arrlist.size();
        answer = new int[n];
        for (int i = 0; i < n; i++){
            answer[i] = arrlist.get(i);
        }
        

        return answer;
    }
}